doc/latex.rst

```diff
@@ -500,19 +472,25 @@ Keys that don't need to be overridden unless in special cases are:
    Default: ``'\\printindex'``
 
 ``'fvset'``
-   Customization of ``fancyvrb`` LaTeX package. The default value of
-   ``'\\fvset{fontsize=\\small}'`` is used to adjust for the large character
-   width of the monospace font, used in code-blocks.  You may need to modify
-   this if you use custom fonts.
+   Customization of ``fancyvrb`` LaTeX package.
 
-   Default: ``'\\fvset{fontsize=\\small}'``
+   The default value is ``'\\fvset{fontsize=auto}'`` which means that the
+   font size will adjust correctly if a code-block ends up in a footnote.
+   You may need to modify this if you use custom fonts:
+   ``'\\fvset{fontsize=\\small}'`` if the monospace font is Courier-like.
+
+   Default: ``'\\fvset{fontsize=auto}'``
 
    .. versionadded:: 1.8
 
    .. versionchanged:: 2.0
-      Due to new default font choice for ``'xelatex'`` and ``'lualatex'``
-      (FreeFont), Sphinx does ``\\fvset{fontsize=\\small}`` also with these
-      engines (and not ``\\fvset{fontsize=auto}``).
+      For ``'xelatex'`` and ``'lualatex'`` defaults to
+      ``'\\fvset{fontsize=\\small}'`` as this
+      is adapted to the relative widths of the FreeFont family.
+
+   .. versionchanged:: 4.0.0
+      Changed default for ``'pdflatex'``. Previously it was using 
+      ``'\\fvset{fontsize=\\small}'``.
```

Comment:
also ``auto`` for platex/uplatex, as they too will use txtt (the monospace font from txfonts package)